src/GHC/All.hs

```diff
@@ -161,17 +161,25 @@ parseModuleEx flags file str = timedIO "Parse" file $ do
             Nothing | file == "-" -> getContentsUTF8
                     | otherwise -> readFileUTF8' file
         str <- pure $ dropPrefix "\65279" str -- remove the BOM if it exists, see #130
-        ppstr <- runCpp (cppFlags flags) file str
         let enableDisableExts = ghcExtensionsFromParseFlags flags
-        dynFlags <- parsePragmasIntoDynFlags baseDynFlags enableDisableExts file ppstr
+        dynFlags <- parsePragmasIntoDynFlags baseDynFlags enableDisableExts file str
         case dynFlags of
           Right ghcFlags -> do
             ghcFlags <- pure $ lang_set ghcFlags $ baseLanguage flags
-            case fileToModule file ppstr ghcFlags of
+            -- Avoid running cpp unless Cpp is enabled (see
+            -- https://github.com/ndmitchell/hlint/issues/1075 as to
+            -- why this is a good idea).
+            srcText <-
+              if Cpp `xopt` ghcFlags
```

Comment:
There is quite a lot of vertical space here. No need for separate lines for then and else to live on their own.